src/app/components/form/form.component.ts

```diff
@@ -250,7 +262,7 @@ export class FormComponent implements OnInit, OnChanges {
         digest: ds.digest
       }});
 
-    if (this.newForm.valid) {
+    if ((!submitNotValid && this.newForm.valid) || (submitNotValid && !this.newForm.valid)) {
```

Comment:
Could this be expressed as `submitNotValid != this.newForm.valid`?

Reply:
changed